### DISCUSSION TITLE
gmxmmpbsa valueerror: too many values to unpack (expected 2)

### DISCUSSION THREAD

**Discussion by mohammedGamila** [Valdes-Tresanco-MS/gmx_MMPBSA#437] [Q&A] [answered]
hello,
i am processing a gmxmmpbsa on a protein-ligand complex and unfor. it keeps showing this message valueerror: too many values to unpack (expected 2), and tehe sequence runs as follows:
[INFO   ] Starting gmx_MMPBSA v1.5.7
[INFO   ] Command-line
  mpirun -np 1 gmx_MMPBSA MPI -O -i mmpbsa.in -cs MD.tpr -ci new_index.ndx -cg 1 13 -ct MDnoPBC.xtc -cp topol2.top -o FINAL_RESULTS_MMPBSA.dat -eo FINAL_RESULTS_MMPBSA.csv

[INFO   ] Checking mmpbsa.in input file...
[INFO   ] Checking mmpbsa.in input file...Done.

[INFO   ] Checking external programs...
[INFO   ] cpptraj found! Using /share/apps/conda_envs/gmxMMPBSA/bin/cpptraj
[INFO   ] tleap found! Using /share/apps/conda_envs/gmxMMPBSA/bin/tleap
[INFO   ] parmchk2 found! Using /share/apps/conda_envs/gmxMMPBSA/bin/parmchk2
[INFO   ] sander found! Using /share/apps/conda_envs/gmxMMPBSA/bin/sander
[INFO   ] Using GROMACS version > 5.x.x!
[INFO   ] gmx found! Using /eb/software/GROMACS/2021.3-foss-2021a-CUDA-11.3.1-PLUMED-2.7.2/bin/gmx
[INFO   ] Checking external programs...Done.

[INFO   ] Building AMBER topologies from GROMACS files...
[INFO   ] Get PDB files from GROMACS structures files...
[INFO   ] Making gmx_MMPBSA index for complex...
[INFO   ] Normal Complex: Saving group 1_13 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_COM.pdb
[INFO   ] No receptor structure file was defined. Using ST approach...
[INFO   ] Using receptor structure from complex to generate AMBER topology
[INFO   ] Normal Receptor: Saving group 1 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_REC.pdb
[INFO   ] No ligand structure file was defined. Using ST approach...
[INFO   ] Using ligand structure from complex to generate AMBER topology
[INFO   ] Normal Ligand: Saving group 13 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_LIG.pdb
[INFO   ] Checking the structures consistency...
[INFO   ]
[INFO   ] Using topology conversion. Setting radiopt = 0...
[INFO   ] Building Normal Complex Amber topology...
  File "/share/apps/conda_envs/gmxMMPBSA/bin/gmx_MMPBSA", line 8, in <module>
    sys.exit(gmxmmpbsa())
  File "/share/apps/conda_envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/app.py", line 98, in gmxmmpbsa
    app.make_prmtops()
  File "/share/apps/conda_envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/main.py", line 548, in make_prmtops
    self.FILES.mutant_receptor_prmtop, self.FILES.mutant_ligand_prmtop) = maketop.buildTopology()
  File "/share/apps/conda_envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/make_top.py", line 118, in buildTopology
    tops = self.gmxtop2prmtop()
  File "/share/apps/conda_envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/make_top.py", line 478, in gmxtop2prmtop
    com_top = self.cleantop(self.FILES.complex_top, self.indexes['COM']['COM'])
  File "/share/apps/conda_envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/make_top.py", line 765, in cleantop
    rtemp_top = parmed.gromacs.GromacsTopologyFile(ttp_file.as_posix())
  File "/share/apps/conda_envs/gmxMMPBSA/lib/python3.9/site-packages/parmed/gromacs/gromacstop.py", line 251, in __init__
    self.read(fname, defines, parametrize)
  File "/share/apps/conda_envs/gmxMMPBSA/lib/python3.9/site-packages/parmed/gromacs/gromacstop.py", line 357, in read
    name, num = line.split()
ValueError: too many values to unpack (expected 2)
Exiting. All files have been retained.

please i need to know how to fix it

**Comment by marioernestovaldes** [comment]
Seems something is wrong with the topology format, usually is one additional space that is ok for GROMACS but not ParmEd. could you please share the files you are using?

**Comment by mohammedGamila** [comment]
thanks alot for your help
here i upload my topol file
again i appreciate your effort
[topol.zip](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/files/13510252/topol.zip)

**Comment by marioernestovaldes** [comment, marked as answer]
in the [ molecules ] section it has:

```
[ molecules ]
; Compound        #mols
Protein_chain_A     1
LIG                 1SOL             16315
SOL         16239
NA               33
CL               43
```

Try deleting the additional `SOL             16315` besides `LIG                 1` and see if that solves the issue

```
[ molecules ]
; Compound        #mols
Protein_chain_A     1
LIG                 1
SOL         16239
NA               33
CL               43
```

**Reply by mohammedGamila** [reply]
greetingd Dr.
this suggestion successfully solved the issue, but i have another issue the process works perfectly in the manual mode but always gives error in the sbatch mode which is:

ValueError: could not convert string to float: ''
Exiting. All files have been retained.
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[55616,1],8]
  Exit code:    1
--------------------------------------------------------------------------

please could you help me in this issue

**Comment by marioernestovaldes** [comment]
I presume you are using an action.to Petron the calculations with sbatch and think this is generating the issue and not gmx_MMPBSA itself. Could you please comment on this?

**Reply by mohammedGamila** [reply]
here i upload the sbatch file which contains only the code
[run_mmgbsa.zip](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/files/13521501/run_mmgbsa.zip)

**Comment by marioernestovaldes** [comment]
Have you tried to run this command on your computer?

**Reply by mohammedGamila** [reply]
the command works perfect but in sbatch gives me errors like i mentioned before

**Comment by marioernestovaldes** [comment]
then, that is something related to the script or batch system and it has nothing to do with gmx_MMPBSA... try using the fewer possible flags in the sbatch script and see how it goes... then start incorporating others until you reach the error

**Reply by mohammedGamila** [reply]
now it gives me this error
MMPBSA_Error: /eb/software/GROMACS/2021.3-foss-2021a-CUDA-11.3.1-PLUMED-2.7.2/bin/gmx trjconv failed when querying MDnoPBC.xtc
Check the gmx_MMPBSA.log file to report the problem.
Exiting. All files have been retained.
[ERROR  ] MMPBSA_Error /eb/software/GROMACS/2021.3-foss-2021a-CUDA-11.3.1-PLUMED-2.7.2/bin/gmx trjconv failed when querying MDnoPBC.xtc.
           Check the gmx_MMPBSA.log file to report the problem.

**Comment by marioernestovaldes** [comment]
can you attach the gmx_MMPBSA.log file?

**Reply by mohammedGamila** [reply]
thanks for your help
i really appreciate it

[gmx_MMPBSA.log](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/files/13533737/gmx_MMPBSA.log)

**Comment by marioernestovaldes** [comment]
Seems.there is a missmatch between your index and your input structure and trajectory. Plese create a new index using the input structure.

If that doesn't solve the problem feel free to send us the files to see what's wrong

**Reply by mohammedGamila** [reply]
thanks dr. mario
here i attach the files 
[New folder (3).zip](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/files/13536869/New.folder.3.zip)
except the MDnoPBC.xtc file which exceeds 2 GB

**Reply by mohammedGamila** [reply]
dear mario
i removed the mpirun part and it worked fine

thank you for your patience and help
